### PR TITLE
Pass task execution metadata from agent

### PR DESCRIPTION
**Pull request by honnix**
# TL;DR
Modify proto to pass task execution metadata from agent to out-core plugin.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [x] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
As described in https://github.com/flyteorg/flyte/issues/3824, for certain type of use cases, the task execution metadata would be desired to have.

## Tracking Issue
https://github.com/flyteorg/flyte/issues/3824

## Follow-up issue
_NA_

### CODE REVIEW

File: protos/flyteidl/admin/agent.proto [hunk 1]
```diff
@@ -16,6 +17,15 @@ enum State {
   SUCCEEDED = 4;
 }
 
+message TaskExecutionMetadata {
```
Comment (hamersaw) [inline protos/flyteidl/admin/agent.proto:20]: docs?

File: protos/flyteidl/admin/agent.proto [hunk 2]
```diff
@@ -36,7 +43,7 @@ message CreateTaskRequest {
   core.TaskTemplate template = 2;
   // Prefix for where task output data will be written. (e.g. s3://my-bucket/randomstring)
   string output_prefix = 3;
-  // subset of task execution metadata that are relevant to out-core plugins
+  // subset of runtime task execution metadata that are relevant to external plugins
```
Comment (honnix) [inline protos/flyteidl/admin/agent.proto:46]: Not sure whether `external` is a more official name.

Comment (hamersaw) [reply protos/flyteidl/admin/agent.proto:46]: I think we officially decided that this feature was going to be called "flyte agents". The notion of "out-of-core plugins" means that nothing will be compiled into propeller, which this is not fixing for existing plugins, and the "external plugins" was not descriptive enough.

Comment (honnix) [reply protos/flyteidl/admin/agent.proto:46]: As I understood, `agent` refers to the routing plugin https://github.com/flyteorg/flyteplugins/tree/master/go/tasks/plugins/webapi/agent, but here it refers to the receiving part or the RPC server. Or maybe I misunderstood that, it is the RPC server being agent.

Comment (pingsutw) [reply protos/flyteidl/admin/agent.proto:46]: Agent is a Python gRPC server that can run certain tasks (BQ, snowflake, etc). agent plugin (https://github.com/flyteorg/flyteplugins/tree/master/go/tasks/plugins/webapi/agent) is actually a gRPC client, which sends the request to the agent.

File: protos/flyteidl/admin/agent.proto [hunk 3]
```diff
@@ -26,6 +43,8 @@ message CreateTaskRequest {
   core.TaskTemplate template = 2;
   // Prefix for where task output data will be written. (e.g. s3://my-bucket/randomstring)
   string output_prefix = 3;
+  // subset of runtime task execution metadata that are relevant to external plugins
```
Comment (pingsutw) [inline protos/flyteidl/admin/agent.proto:46]:
```suggestion
  // subset of runtime task execution metadata.
```

**Review by pingsutw — APPROVED** [review]

**Comment by honnix** [comment]
@pingsutw Could you please help merge this? Thanks.

**Review by hamersaw — APPROVED** [review]

**Pull request merged**